[AWS Athena] Move From Beta to GA

MichaelYochpaz opened the pull request:
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-5829)

## Description
Move AWS Athena from beta to GA, add new `Timeout` and `Retries` instance configuration parameters, and update the context output.

MichaelYochpaz commented on Packs/AWS-Athena/pack_metadata.json:
```diff
@@ -1,8 +1,8 @@
     "description": "Amazon Web Services Athena",
     "support": "xsoar",
-    "currentVersion": "1.0.8",
+    "currentVersion": "1.1.0",
```
Should we make it a minor bump or a major bump? (2.0.0)

DeanArbel replied:
Let's make it a major bump.

MichaelYochpaz commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.yml:
```diff
@@ -1,56 +1,83 @@
+  name: retries
+  type: 0
+  defaultvalue: 5
   required: false
```
Since we're changing the ID of this integration (which will force customers to configure a new instance anyway), I removed the `access_key` and `secret_key` fields which are replaced by the type-9 `credentials` field and are kept for BC.

MichaelYochpaz commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py:
```diff
@@ -44,205 +19,180 @@ def default(self, obj):
-        kwargs.update({'ResultConfiguration': {'EncryptionConfiguration': {'KmsKey': args.get('KmsKey')}}})
-    if args.get('WorkGroup') is not None:
-        kwargs.update({'WorkGroup': args.get('WorkGroup')})
+def parse_rows_response(rows_data: list[dict]) -> list[dict]:
+    """
+    Parse and arrange the 'Rows' data from the response.
+
+    Args:
+        rows_data (list[dict]): The 'Rows' data from the response.
+
+    Returns:
+        list[dict]: The data in a parsed and arranged format.
+    """
+    keys: list[str] = [item['VarCharValue'] for item in rows_data[0]['Data']]
+    raw_results = [item['Data'] for item in rows_data[1:]]
+    result_data = []
+
+    for raw_result in raw_results:
+        current_item_data = {}
+
+        for idx, value in enumerate(raw_result):
+            if 'VarCharValue' in value:
+                current_item_data[keys[idx]] = value['VarCharValue']
+
+        result_data.append(current_item_data)
+
+    return result_data
```
Example for such a response can be seen on:
https://github.com/demisto/content/pull/30694/files#diff-40ceff67b244e3d04a82f5a5018a9d8781b6bcf2d8c230f632757af55a5eca09
(first item is keys, and then after that, each item is the values for these keys in corresponding indexes. Like a CSV table).

DeanArbel commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py:
```diff
@@ -44,205 +19,180 @@ def default(self, obj):
+    Returns:
+        list[dict]: The data in a parsed and arranged format.
+    """
+    keys: list[str] = [item['VarCharValue'] for item in rows_data[0]['Data']]
```
Is it possible the rows_data will be empty, or is it safe to assume rows_data[0]['Data'] will always be populated with items that have `VarCharValue`?

DeanArbel replied:
?

DeanArbel reviewed (approved):
Very nice work, see my comments.

DeanArbel commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py:
```diff
@@ -44,205 +19,200 @@ def default(self, obj):
+        result: str | CommandResults
+
+        if demisto.command() == 'test-module':
+            response = client.list_named_queries()
+            if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+                result = 'ok'
+
+            else:
+                result = CommandResults(readable_output=f'Error: {response}')
```
Please extract to a function

DeanArbel commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py:
```diff
@@ -44,205 +19,200 @@ def default(self, obj):
-        roleArn=args.get('roleArn'),
-        roleSessionName=args.get('roleSessionName'),
-        roleSessionDuration=args.get('roleSessionDuration'),
+        return_error('Could not parse the received response.')
+        demisto.debug(f'Error:\n{e}\n'
+                      f'Response:\n{raw_response}')
```
The line after `return_error` is unreachable since it exits.
Please fix it and raise an exception instead of calling return_error.

DeanArbel commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py:
```diff
@@ -44,205 +19,200 @@ def default(self, obj):
-    elif demisto.command() == 'aws-athena-get-query-results':
-        get_query_results_command(demisto.args())
+    except Exception as e:
+        return_error(f'Error: {e}')
+        demisto.error(f'Error: {e}\n\nTraceback:\n{traceback.format_exc()}')
```
You don't need the traceback anymore, in any case it's unreachable, please remove.

ilappe commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py:
```diff
@@ -1,248 +1,299 @@
-        response = client.list_named_queries()
-        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
-            demisto.results('ok')
+@polling_function(
```
it's look like `hide_polling_output` is missed in yml
see the documentation of `polling_function`

ilappe commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.yml:
```diff
@@ -2,35 +2,56 @@ commonfields:
   id: AWS - Athena - Beta
   version: -1
 name: AWS - Athena - Beta
```
should `Beta` remaining?

MichaelYochpaz replied:
Yes, we discussed this and decided we want to keep the ID and name (the name must match the ID) like that, and just change the display name, as those are transparent to customers, and changing the ID will require all existing customers to reconfigure a new instance (it will be detected as a new integration).

ilappe reviewed (changes requested):
very clean and elegant!
please see my few comments

MichaelYochpaz commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.yml:
```diff
@@ -114,7 +295,110 @@ script:
+    - contextPath: AWS.Athena.Query.Status.AthenaError.ErrorMessage
+      description: Contains a short description of the error that occurred.
+      type: String
+    - contextPath: AWS.Athena.Statistics.EngineExecutionTimeInMillis
```
You're right, that's a miss from my part, should be under `Query`. I'll fix that :)

MichaelYochpaz commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py:
```diff
@@ -1,248 +1,299 @@
+                                     kms_key=args.get('KmsKey'), work_group=args.get('WorkGroup'))
+
+    context_data = {
+        'QueryString': query_string,
```
This is the dictionary I push to context using `outputs_prefix=f'AWS.Athena.{QUERY_DATA_OUTPUTS_KEY}',`.
So the final context path would be `AWS.Athena.Query.QueryString` (this specific item is a string containing the SQL query that was used in the command)

ilappe replied:
@MichaelYochpaz,yes, but the key in the yml are `AWS.Athena.Query.Query` :)

MichaelYochpaz replied:
You are right, it was changed while I was working on it and I forgot to update it.
Should be `Query` to match the key that's returned by the API when running `aws-athena-get-query-execution` (to avoid redundant keys with different names).
I pushed a fix :)

MichaelYochpaz commented on Packs/AWS-Athena/TestPlaybooks/playbook-AWS-Athena-Beta.yml:
```diff
@@ -1,91 +0,0 @@
-id: Beta-Athena-Test
-version: -1
-name: Beta-Athena-Test
```
We removed the test-playbook for now until [CIAC-8849](https://jira-dc.paloaltonetworks.com/browse/CIAC-8849) is resolved. Once it's resolved, there's [CIAC-8850](https://jira-dc.paloaltonetworks.com/browse/CIAC-8850) to re-add the playbook and update it.

MichaelYochpaz commented on Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py:
```diff
@@ -1,248 +1,299 @@
+
+        for idx, value in enumerate(raw_result):
+            if 'VarCharValue' in value:
+                current_item_data[keys[idx]] = value['VarCharValue']
```
I'll add a comment to the docstring

MichaelYochpaz commented:
@ilappe Thanks for the review. I addressed the relevant notes on https://github.com/demisto/content/pull/30694/commits/02025a828d83477596018baf2545ad8bbe98a4e0, and added context to the other notes.
Should be ready for a re-review.

DeanArbel reviewed (approved):
Very impressive work. Good job

ilappe reviewed (approved):
Great work!
